### PR TITLE
add option status code log from origin

### PR DESCRIPTION
**Pull request by t4niwa**
# Description

add option status code log from origin

## Type of change

- [ ] Bug fix
- [x] New feature
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Non-code changes (update documentation, pipeline, etc.)

### Flags

- [ ] Breaks backward compatibility
- [ ] Requires a documentation update
- [ ] Has untestable code

## Related issue/PR

**Delete this section if there are no issues or pull requests that relate to this pull request.**
- Fixes #_issue_
- Closes #_PR_

---

## Checklist

- [x] Followed the guidelines in the CONTRIBUTING document
- [x] Added prefix `[skip ci]`/`[ci skip]`/`[no ci]`/`[skip actions]`/`[actions skip]` in the PR title if necessary
- [x] Tested and linted the code
- [x] Commented the code
- [x] Made corresponding changes to the documentation
- [x] Passed all pipeline checking

## Checklist for maintainer
- [ ] Use `Squash and merge`
- [ ] Double-confirm the merge message has prefix `[skip ci]`/`[ci skip]`/`[no ci]`/`[skip actions]`/`[actions skip]`
- [ ] Delete the branch after merge

### CODE REVIEW

File: test/data/example_config.yaml [hunk 1]
```diff
@@ -44,6 +44,10 @@ proxy:
     writeBufferSize: 0
     readBufferSize: 0
     forceAttemptHTTP2: true
+  originLog:
+    statusCode:
+      enable: true
+      exclude: [200]
```
Comment (WindzCUHK) [inline test/data/example_config.yaml:50]:
```suggestion
      exclude:
        - 200
```

File: handler/handler.go [hunk 2]
```diff
@@ -71,6 +71,17 @@ func New(cfg config.Proxy, bp httputil.BufferPool, prov service.Authorizationd)
...
+			if cfg.OriginLog.StatusCode.Enable {
+				for _, statusCode := range cfg.OriginLog.StatusCode.Exclude {
+					if statusCode == res.StatusCode {
+						return nil
+					}
+				}
+				glg.Infof("Origin request: %s %s, Response: status code: %d", res.Request.Method, res.Request.URL, res.StatusCode)
+			}
+			return nil
+		},
```
Comment (WindzCUHK) [inline handler/handler.go:84]:
```suggestion
		ModifyResponse: modifyResponse,
```

maybe better to define the function before `&httputil.ReverseProxy{}`?
```go
modifyResponse := func(res *http.Response) error {
    ...
}
if !cfg.OriginLog.StatusCode.Enable {
    modifyResponse = nil
}
```

**Review by WindzCUHK — CHANGES_REQUESTED** [review]

**Comment by WindzCUHK** [comment]
@t4niwa 
Can help to add a unit test like below?
https://github.com/AthenZ/authorization-proxy/blob/93fe09aadb4210fd7f8ea1eb9a4427377a69d05c/handler/handler_test.go#L519-L536

Just check when `cfg.OriginLog.StatusCode.Enable == true`, `got.ModifyResponse != nil` is OK.

**Review by ssunorz — APPROVED** [review]

**Comment by kyfujisa** [comment]
↑の指摘事項への対応が行われたことを確認しました。

**Review by kyfujisa — APPROVED** [review]

**Review by WindzCUHK — APPROVED** [review]

**Pull request merged**